File: Practica-Multilenguaje/Python/palindromo.py
import re

def es_palindromo(texto):
    texto_limpio = re.sub(r'[^a-zA-Z0-9]', '', texto.lower())
    return texto_limpio == texto_limpio[::-1]

if __name__ == "__main__":
    cadena = "Anita lava la tina"
    if es_palindromo(cadena):
        print("Es un palíndromo")
    else:
        print("No es un palíndromo")

File: Practica-Multilenguaje/cpp/palindromo.cpp
#include <iostream>
#include <algorithm>
#include <cctype>

bool esPalindromo(std::string texto) {
    std::string limpio = "";
    for (char c : texto) {
        if (isalnum(c)) limpio += tolower(c);
    }
    std::string reverso = limpio;
    std::reverse(reverso.begin(), reverso.end());
    return limpio == reverso;
}

int main() {
    std::string cadena = "Anita lava la tina";
    if (esPalindromo(cadena)) {
        std::cout << "Es un palíndromo" << std::endl;
    } else {
        std::cout << "No es un palíndromo" << std::endl;
    }
    return 0;
}